### PR TITLE
compatible(integration_test_charm.yaml): Add `microk8s` cloud

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**

### CODE REVIEW

**Comment by deusebio** [comment]
fine with the changes but this action seems to be doing very similar things as `charmed-kubernetes/actions-operator@main` , namely setting up an env with either lxd or microk8s to run juju

**Comment by deusebio** [comment]
Uhm, the reason I'm saying this is because the PR and the workflow are mostly concerned with setting up envs, which is really not db specific. I suppose it would make more sense to use the "official" action (of course, if the upstream is maintained by someone and it is not deprecated) for taking care of setting up the envs, such that this will be maintained in a centralized way, and we don't divert from it. 

If the original action does not support some feature, I believe it could be beneficial for the community to extend the action with these by contributing upstream. 

I have the feeling that the amount of logic we are starting to put in the action which is NOT data platform specific starts to be significant, and in general something other people (outside of DP) generally needs as well

**Comment by deusebio** [comment]
Anyhow, if this PR is blocking other things you are working on in SQL, feel free to go ahead. As said, I don't have concerns around the code itself, but around whether we should refactor it to use more the official action. 

This is however outside of the scope of the PR, but I would give a strong nudge to consider it

**Review by deusebio — APPROVED** [review]

File: .github/workflows/integration_test_charm.yaml [hunk 1]
```diff
@@ -124,10 +131,29 @@ jobs:
...
+          CLOUD = "${{ inputs.cloud }}"
+          if CLOUD == "lxd":
+              group = "lxd"
+          elif CLOUD == "microk8s":
+              SNAP_CHANNEL = "${{ inputs.microk8s-snap-channel }}"
+              assert (
+                  SNAP_CHANNEL != ""
+              ), '`microk8s-snap-channel` input required if `cloud` is "microk8s"'
+              assert "strict" in SNAP_CHANNEL.lower(), "Only strict microk8s snap supported"
+              group = "snap_microk8s"
```
[the remark this answers is not in the document]
Comment (juditnovak) [reply .github/workflows/integration_test_charm.yaml:150]: Responded here: https://chat.canonical.com/canonical/pl/rxb6as8h4jfsbke6ku7jzz7xqe

**Review by juditnovak — APPROVED** [review]
Nice :-)

**Pull request merged**